test/heapwatch/metrics_lib.py

```diff
@@ -210,14 +210,15 @@ def add_tag(self, key: str, value: str):
         self.tags[key] = value
         self.tag_keys.add(key)
 
-    def has_tags(self, tag_keys: set, tags: Dict[str, str]):
+    def has_tags(self, tags: Dict[str, Tuple[str, ...]], tag_keys: Set[str] | None) -> bool:
         """return True if all tags are present in the metric tags
         tag_keys are not strictly needed but used as an optimization
         """
-        if self.tag_keys.intersection(tag_keys) != tag_keys:
+        if tag_keys is not None and self.tag_keys.intersection(tag_keys) != tag_keys:
             return False
-        for k, v in tags.items():
-            if self.tags.get(k) != v:
+        for k, vals in tags.items():
+            v = self.tags.get(k)
+            if v not in vals:
```

Comment:
Why is this check necessary, aren't we iterating through the collection grabbing the a values collection for that key already?

Reply:
it checks if a metric has key-value matching to at least one value from the list provided - like if metrics has `proto: "/meshsub"` label and the filter is `proto: ["/meshsub", "/algorand-ws"]`.